editor/store/selectors.js

```diff
@@ -1435,3 +1435,37 @@ export function isPublishingPost( state ) {
 export function getProvisionalBlockUID( state ) {
 	return state.provisionalBlockUID;
 }
+
+/**
+ * Returns whether the permalink is editable or not.
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {boolean} Whether or not the permalink is editable.
+ */
+export function isPermalinkEditable( state ) {
+	const samplePermalinkData = getEditedPostAttribute( state, 'sample_permalink' );
+
+	if ( ! samplePermalinkData || samplePermalinkData.length < 2 ) {
+		return false;
+	}
+
+	return /%(?:postname|pagename)%/.test( samplePermalinkData[ 0 ] );
+}
+
+/**
+ * Returns the sample permalink for the post.
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {string} The sample permalink.
+ */
+export function getSamplePermalink( state ) {
+	const samplePermalinkData = getEditedPostAttribute( state, 'sample_permalink' );
+
+	if ( ! samplePermalinkData || samplePermalinkData.length < 2 ) {
+		return false;
```

Comment:
There's already two other value types to represent an empty value: `null` and `undefined`. We needn't return a boolean here. Elsewhere we return `null` for the empty early return, which works well with [JSDoc's nullable type](http://usejsdoc.org/tags-type.html).

Maybe just copy-pasta from `isPermalinkEditable` ?